test/ClangImporter/incomplete_objc_types_importing_limitations_diagnostics.swift

```diff
@@ -0,0 +1,25 @@
+// RUN: not %target-swift-frontend -enable-import-objc-forward-declarations -enable-objc-interop -typecheck -I %S/Inputs/custom-modules/IncompleteTypes %s 2>&1 | %FileCheck %s
+// RUN: not %target-swift-frontend -swift-version 6 -enable-objc-interop -typecheck -I %S/Inputs/custom-modules/IncompleteTypes %s 2>&1 | %FileCheck  %s
+
+// REQUIRES: objc_interop
+
+import IncompleteTypeLibrary1
+
+let incompleteInterface = CFunctionReturningAForwardDeclaredInterface1()!
+let incompleteProtocol = CFunctionReturningAForwardDeclaredProtocol1()!
+
+incompleteInterface.doSomethingForwardDeclaredInterfacesCan()
+// CHECK: incomplete_objc_types_importing_limitations_diagnostics.swift:{{[0-9]+}}:{{[0-9]+}}: error: value of type 'ForwardDeclaredInterface' has no member 'doSomethingForwardDeclaredInterfacesCan'
+// CHECK: incompleteInterface.doSomethingForwardDeclaredInterfacesCan()
+// CHECK: <unknown>:0: note: class 'ForwardDeclaredInterface' is a placeholder for a forward declared Objective-C interface and may be missing members; import the definition to access the complete interface
```

Comment:
`<unknown>:0` looks a little unfortunate, but there is no source location available for a synthesized type.